contracts/peg/curve/CurveRouter.sol

```diff
@@ -0,0 +1,154 @@
+    function getRedeemAmountOut(
+        uint256 amountVoltIn,
+        address tokenA,
+        address tokenB
+    ) external view override returns (uint256 amountOut) {}
+
+    // ---------- State-Changing API ----------
+
+    /// @notice Mint volt for stablecoins via curve
+    /// @param to, the address to mint Volt to
+    /// @param psm, the PSM the router should mint from
+    /// @param tokenA, the inital token that the user would like to swap
+    /// @param tokenB, the token the user would route through
+    /// @return amountOut the amount of Volt returned from the mint function
+
+    function mint(
+        address to,
+        uint256 amountIn,
+        IPegStabilityModule psm,
+        address curvePool,
+        address tokenA,
+        address tokenB
+    ) external override returns (uint256 amountOut) {
+        (
+            uint256 amountTokenBReceived,
+            uint256 amountVoltOut,
+            int256 index_i,
+            int256 index_j
+        ) = _calculateSwap(amountIn, psm, curvePool, tokenA, tokenB);
+
+        IERC20(tokenA).transferFrom(msg.sender, address(this), amountIn);
+
+        IERC20(tokenA).approve(address(curvePool), type(uint256).max);
+
+        ICurvePool(curvePool).exchange(
+            int128(index_i),
+            int128(index_j),
+            amountIn,
+            amountTokenBReceived
+        );
+
+        amountOut = psm.mint(to, amountTokenBReceived, amountVoltOut);
+    }
+
+    /// @notice Redeems volt for stablecoin via curve
+    /// @param to, the address to send redeemed stablecoin to
+    /// @param amountVoltIn, the amount of VOLT to deposit
+    /// @param minAmountOut, the minimum amount of stablecoin expected to be received
+    /// @param psm, the PSM the router should redeem from
+    /// @param tokenA, the token to route through on redemption
+    /// @param tokenB, the token the user would like to redeem
+    /// @return amountOut the amount of stablecoin returned from the mint function
+    function redeem(
+        address to,
+        uint256 amountVoltIn,
+        uint256 minAmountOut,
+        address psm,
+        address tokenA,
+        address tokenB
+    ) external override returns (uint256 amountOut) {}
+
+    // ---------- Private functions ----------
+
+    function _calculateSwap(
+        uint256 amountIn,
+        IPegStabilityModule psm,
+        address curvePool,
+        address tokenA,
+        address tokenB
+    )
+        private
+        view
+        returns (
+            uint256 amountTokenBReceived,
+            uint256 amountVoltOut,
+            int256 index_i,
+            int256 index_j
+        )
+    {
+        require(
+            address(psm.underlyingToken()) == tokenB,
+            "CurveRouter: Unsupported Token"
+        );
+
+        // todo to be able to search up to 4 tokens -- curve reverts the call when accessing an index that doesn't exist
+        // maybe pass number of tokens in the pool in the function
+
```

Comment:
Todo: to be able to search up to 4 tokens, as some curve pools have 4 tokens -- curve reverts the call when accessing an index that doesn't exist

maybe pass number of tokens in the pool in the function